storage/engine/rocksdb/db.cc

```diff
@@ -899,9 +899,9 @@ struct IteratorGetter : public Getter {
 struct DBGetter : public Getter {
   rocksdb::DB *const rep;
   rocksdb::ReadOptions const options;
-  rocksdb::Slice const key;
+  std::string const key;
 
-  DBGetter(rocksdb::DB *const r, rocksdb::ReadOptions opts, rocksdb::Slice k)
+  DBGetter(rocksdb::DB *const r, rocksdb::ReadOptions opts, const std::string &k)
```

Comment:
Nit: this argument could still be a slice. The important part is that the `key` member makes a copy. If a caller had a slice it wouldn't make sense for it to copy the slice to a string to pass into `DBGetter` if `DBGetter` is just going to copy it again.